src/main/java/net/dries007/tfc/common/entities/ai/BrainObjects.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.entities.ai;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import net.minecraft.world.entity.ai.memory.MemoryModuleType;
+import net.minecraft.world.entity.schedule.Activity;
+import net.minecraft.world.entity.schedule.Schedule;
+import net.minecraft.world.entity.schedule.ScheduleBuilder;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.ForgeRegistries;
+import net.minecraftforge.registries.RegistryObject;
+
+import com.mojang.serialization.Codec;
+import net.dries007.tfc.TerraFirmaCraft;
+
+public class BrainObjects
+{
+    public static final DeferredRegister<Activity> ACTIVITIES = DeferredRegister.create(ForgeRegistries.ACTIVITIES, TerraFirmaCraft.MOD_ID);
+    public static final DeferredRegister<MemoryModuleType<?>> MEMORY_TYPES = DeferredRegister.create(ForgeRegistries.MEMORY_MODULE_TYPES, TerraFirmaCraft.MOD_ID);
+    public static final DeferredRegister<Schedule> SCHEDULES = DeferredRegister.create(ForgeRegistries.SCHEDULES, TerraFirmaCraft.MOD_ID);
+
+    public static final Activity HUNT = registerActivity("hunt");
+
+    public static final RegistryObject<Schedule> DIURNAL = registerSchedule("diurnal", newSchedule().changeActivityAt(0, HUNT).changeActivityAt(11000, Activity.REST)::build);
+    public static final RegistryObject<Schedule> NOCTURNAL = registerSchedule("nocturnal", newSchedule().changeActivityAt(0, Activity.REST).changeActivityAt(11000, HUNT)::build);
```

Comment:
This looks convoluted. I *think* you should be able to rely on since Activity is sorted before Schedule, that you can unbox the activity registry object in the supplier. **But** only if you change it to invoke
```java
() -> newSchedule().foo()....build()
```
as that won't invoke `newSchedule()` until after.

I think this blows up in 1.18.2 anyway, because the actual instance needs to be created during register events (or so I've been told).